[Quah Kian Seong] iP

kianseong opened the pull request:
### DukePro
>&ensp;"Your mind is for having ideas, not holding them." - David Allen [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Project Setup: Test Commit

Commit:
Implemented Duke Level-1

Commit:
Implemented Duke Level-2

Commit:
Implemented Duke Level-3

Commit:
Implemented Duke Level-4

Commit:
Implemented A-TextUiTesting. Updated runtest.sh to be compatible with fish shell.

Commit:
Implemented Duke Level-5.

Commit:
Implemented Duke Level-6

Commit:
Implemented A-Enums
Moved driver to Main.java
Created duke package (default)

Commit:
Moved tasks management to TaskManager
Implemented enums for different task types

Bug Fixes:
A task could not be marked/unmarked/deleted if there was only 1 task in the list.

Commit:
Cleaned up code and fixed bugs

Commit:
Updated test cases and runtest.sh

Commit:
Implemented Level-7: Save

Commit:
Implemented Level-8: Dates and Times

Commit:
Implement A-MoreOOP: Use More OOP

Commit:
Implement A-Packages: Organize into Packages

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'
Merge add-gradle-support

Commit:
Implement A-Gradle: Use Gradle

Commit:
Implement A-JUnit: Add JUnit Tests

Commit:
Implement A-JavaDoc

Commit:
Implement A-CodingStandard

Commit:
Implement Level-9: Find

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-Level-9'

Commit:
Add colour support with new command colour

Commit:
Clean up code

Commit:
add a barebones GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
added some usage of varargs

Commit:
Merge branch 'branch-A-Varargs'

Commit:
cleaned up code according to CheckStyle

Commit:
Merge branch 'branch-A-CheckStyle'

kangzongxian commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,204 @@
+        return parseArgsAsDescriptionAndDate(EVENT_FORMAT, args);
+    }
+
+    private int parseArgsAsDisplayedIndex(String args) throws DukeException {
```
Could we perhaps add JavaDocs for the methods so that we can make things a little bit clearer?

kangzongxian commented on src/main/java/duke/storage/StorageFile.java:
```diff
@@ -0,0 +1,114 @@
+     * Loads the {@code TaskList} data from the storage file and return it.
+     * Returns an empty arraylist of Task if the file does not exist.
+     */
+    public List<Task> load() {
```
Should we maybe add comments here as well since the code is quite a large chunk, so that it may be easier to understand?

kangzongxian commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,64 @@
+package duke;
```
Overall, I think it is a really good job, and I especially like the separation of commands to make each file much more manageable!

chanyuk replied:
I agree! You have clearly abstracted out the different classes with different functions.

kangzongxian commented on src/main/java/duke/storage/StorageFile.java:
```diff
@@ -0,0 +1,114 @@
+            String description;
+
+            for (String line : contents) {
+                inputArray = Arrays.stream(line.split("\\|")).map(String::trim).toArray(String[]::new);
```
Should we perhaps give a brief comment about the code here so that there is better clarity with multiple methods involved?

kangzongxian commented on src/main/java/duke/commands/UnmarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @Override
+    public CommandResult execute(TaskList taskList, TextUi textUi, StorageFile storage) {
```
Should we perhaps add JavaDocs here as well, since there are multiple different parts being involved here?

chanyuk commented on src/main/java/duke/DukeApp.java:
```diff
@@ -0,0 +1,137 @@
+    }
+
+    /**
+     * Respond to user input.
```
I think you could try following the JavaDoc format for some of these functions?

```suggestion
     * Responds to user input.
     *
     * @param input User input to respond to.
     * @return String response to input
```

chanyuk commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,28 @@
+     * @param taskList task list manager
+     * @param textUi manages Duke's interaction with the user
+     * @param storage manages the storage of Duke's data
+     */
```
I think you might be missing a returns section in the JavaDoc here?
```suggestion
     * @return ...
     */
```

chanyuk reviewed (commented):
Overall looks good! Just some minor suggestions about JavaDocs.

agarwal-anjali commented on src/main/java/duke/commands/CommandResult.java:
```diff
@@ -0,0 +1,17 @@
+
+    private final String feedbackToUser;
+
+    public CommandResult(String feedbackToUser) {
+        this.feedbackToUser = feedbackToUser;
+    }
+
+    public String getFeedbackToUser() {
+        return feedbackToUser;
+    }
```
As these 2 methods are public, may be you can consider adding JavaDocs for them.

agarwal-anjali commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class DeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
```
May be you can consider adding JavaDoc for these public class members!

agarwal-anjali commented on src/main/java/duke/commands/HelpCommand.java:
```diff
@@ -0,0 +1,33 @@
+public class HelpCommand extends Command {
+
+    public static final String COMMAND_WORD = "help";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Shows this help page.\n"
+            + "\tEx.: " + COMMAND_WORD;
```
I liked the way you used MESSAGE_USAGE to print the help commands!

agarwal-anjali commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,204 @@
+/**
+ * Parses user input.
+ */
+public class Parser {
```
To make your code easy to follow and readable, may be you can consider adding JavaDocs for all the functions in this class.

agarwal-anjali commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,117 @@
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
-
+	────────────────────────────────────────────────────────────
```
Your output looks well structured!👍

agarwal-anjali reviewed (commented):
Your code looks well structured and it seems you have made a good use of OOP principles. As a suggestion may be you can just consider adding a few comments to make your code more readable!

Commit:
Clean up code

Rename Duke to Sus

Commit:
add gradle workflow

Commit:
fix directory of gradle workflow

Commit:
add one assertion

Commit:
improve code quality

Commit:
indicate the use of streams

kianseong committed:
Merge pull request #2 from KIANSEONG/branch-A-Assertions

add one assertion

Commit:
Merge branch 'master' into branch-A-Streams

Commit:
Merge branch 'master' into branch-A-CodeQuality

kianseong committed:
Merge pull request #3 from KIANSEONG/branch-A-Streams

indicate the use of streams

Commit:
Merge branch 'master' into branch-A-CodeQuality

kianseong committed:
Merge pull request #4 from KIANSEONG/branch-A-CodeQuality

improve code quality

Commit:
fix junit tests

Commit:
add ability to update the description of a task

clean up some code

Commit:
Merge branch 'branch-C-Update'

Commit:
improve code quality

make GUI text bold for readability

add message to inform user if date input is not of correct format

make error messages more descriptive

Commit:
build: remove third party library Jansi and replace with ANSI colour codes

Commit:
style: change some messages

Commit:
fix: fix colour command not checking arguments

Commit:
docs: add a user guide

Commit:
build: update build and test files

Commit:
build: change permission for gradlew

Commit:
build: change permission for runtest.sh

Commit:
build: change permission for gradlew

Commit:
Merge branch 'master' of https://github.com/KIANSEONG/nus-CS2103T-ip

Commit:
feat: update storage file path

Storage file now saves to working directory.

Commit:
Release the product to be used

Commit:
docs: update user guide
